src/context/provider-context.js

```diff
@@ -0,0 +1,52 @@
+import axios from "axios";
+import React, { createContext, useContext, useEffect, useReducer } from "react";
+import { reducer } from "../reducer/reducer";
+
+export const ProviderContext = createContext();
+
+export const useProvider = () => useContext(ProviderContext);
+
+const initialState = {
+  products: { products: [] },
+  categories: {},
+  filters: {
+    categories: [],
+    filterRating: null,
+    priceUpperRange: null,
+    sortBy: null,
+  },
+  cart: {
+    products: [],
+    decrement: false,
+  },
+  wishlist: { products: [] },
+};
+
+export const Provider = ({ children }) => {
```

Comment:
I think it can be named better, rather than just Provider and ProviderContext, you should use a more self-explaining name, such as CategoriesProvider or ProductProvider. Since in future you can add more Providers like AuthProvider and ThemeProvider in future

Reply:
Done.